GitUI/CommandsDialogs/FormRemotes.cs

```diff
@@ -255,8 +272,13 @@ private void InitialiseTabBehaviors()
             RemoteBranches.DataSource = heads;
             RemoteBranches.ClearSelection();
             RemoteBranches.SelectionChanged += RemoteBranchesSelectionChanged;
-
-            if (RemoteBranches.Rows.Count > 0)
+            var preselectLocalRow = RemoteBranches.Rows.Cast<DataGridViewRow>().
+                FirstOrDefault(r => r.DataBoundItem is IGitRef gitRef ? gitRef.LocalName == preselectLocal : false);
+            if (preselectLocalRow != null)
+            {
+                preselectLocalRow.Selected = true;
```

Comment:
The selection of the line doesn't work for me. The first line is always the one selected.
And I don't know why 😢  Because the code is not made in UI thread?

We should find a way to fix it.

Reply:
I found it why and fixed it also in my branch.